sklearn/decomposition/kernel_pca.py

```diff
@@ -54,10 +62,14 @@ class KernelPCA(BaseEstimator, TransformerMixin):
         Learn the inverse transform for non-precomputed kernels.
         (i.e. learn to find the pre-image of a point)
 
-    eigen_solver : string ['auto'|'dense'|'arpack'], default='auto'
-        Select eigensolver to use. If n_components is much less than
-        the number of training samples, arpack may be more efficient
-        than the dense eigensolver.
+    eigen_solver : string ['auto'|'dense'|'arpack'|'randomized'],
+        default='auto'
+        Select eigensolver to use. If n_components is much less than the number
+        of training samples, randomized (or arpack to a smaller extend) may be
+        more efficient than the dense eigensolver. randomized SVD is done
+        according to the method of Halko et al.
```

Comment:
Could the "auto" selection rule be explained more explicitly here? The docstring of the PCA gives more details:

http://scikit-learn.org/stable/modules/generated/sklearn.decomposition.PCA.html